msal-client-credentials-flow/README.md

```diff
@@ -0,0 +1,163 @@
+| **Name**                          | `Node Web API`                                                               | Suggested value for this sample. <br/> You can change the app name at any time.                    |
+| **Supported account types**       | **Accounts in this organizational directory only (Single tenant)**           | Suggested value for this sample.                                                                   |
+| **Platform type**                 | _None_                                                                       | No redirect URI required; don't select a platform.                                                                    |
+| **Client secret**                 | _**Value** of the client secret (not its ID)_                                | :warning: Record this value immediately! <br/> It's shown only _once_ (when you create it).        |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+```javascript
+auth: {
+  // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+  clientId: '',
+
+  // Client secret 'Value' (not the ID) from 'Client secrets' in app registration in Azure portal
+  clientSecret: '',
+
+  // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+  authority: ''
+  
+  // 'Object ID' of app registration in Azure portal - this value is a GUID
+  clientObjectId: ''
+}
+```
+
+### 3. Install package(s)
+
+To install Node.js and MSAL libraries:
+
+```bash
+npm install
+```
+
+## Run the application
+
+```bash
+node app.js
+```
+
+If everything worked, you should receive a response from the API similar to this:
+
+```console
+$ curl http://localhost:8080/
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#applications/$entity",
+  "id": "003c4673-d4a0-4eb6-8979-d6269e08b87e",
+  "deletedDateTime": null,
+  "appId": "c085fc22-2484-41c0-b795-2e940f51df64",
+  "applicationTemplateId": null,
+  "disabledByMicrosoftStatus": null,
+  "createdDateTime": "2021-12-17T16:45:43Z",
+  "displayName": "Node Web API",
+  "description": null,
+  "groupMembershipClaims": null,
+  "identifierUris": ["api://c085fc22-2484-41c0-b795-2e940f51df64"],
+  "isDeviceOnlyAuthSupported": null,
+  "isFallbackPublicClient": true,
+  "notes": null,
+  "publisherDomain": "contoso.onmicrosoft.com",
+  "serviceManagementReference": null,
+  "signInAudience": "AzureADMyOrg",
+  "tags": [],
+  "tokenEncryptionKeyId": null,
+  "defaultRedirectUri": null,
+  "certification": null,
+  "optionalClaims": null
+  …
+}
+```
+
+## About the code
+
+This Node.js web API has a single route that supports anonymous access. When the anonymous route is called, the API requests its own application object from Microsoft Graph.
```

Comment:
Two thoughts:

- Name of route/endpoint?
- Express isn't mentioned here or in the intro - seems that's something worth mentioning, no?